### DISCUSSION TITLE
lc/128/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#2735] [Announcements]
# lc/128/

多种编程语言实现 LeetCode、《剑指 Offer（第 2 版）》、《程序员面试金典（第 6 版）》题解

https://leetcode.doocs.org/lc/128/

**Comment by lazyJLBL** [comment]
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)  # 将列表转换为集合，以提高查找速度
        maxcount = 0
        for num in s:
            if num - 1 not in s:  # 只检查递增序列的起始元素
                count = 1
                current_num = num
                while current_num + 1 in s:  # 继续向后查找递增元素
                    count += 1
                    current_num += 1
                maxcount = max(maxcount, count)
        return maxcount
这么写会更快一些

**Reply by lazyJLBL** [reply]
这么写就不用每次while都判断一次，会比之前的快许多

**Reply by lazyJLBL** [reply]
```python3
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)  # 将列表转换为集合，以提高查找速度
        maxcount = 0
        for num in s:
            if num - 1 not in s:  # 只检查递增序列的起始元素
                count = 1
                current_num = num
                while current_num + 1 in s:  # 继续向后查找递增元素
                    count += 1
                    current_num += 1
                maxcount = max(maxcount, count)
        return maxcount
```

**Reply by yanglbme** [reply]
跟方法二的写法差不多

**Reply by lazyJLBL** [reply]
是滴，这个吧比较放在了while循环的外面，这样就更快了
